### PR TITLE
feat: add --normalize-schema in cli

### PR DESCRIPTION
**Pull request by Keith-CY**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,161 @@
...
+vector Words <Word>;
+```
+
+There're two steps to normalize it.
+
+#### 1. Use [`Moleculec`](https://github.com/nervosnetwork/molecule/) to transform Molefile into JSON file, which is valid in JavaScript
+
+This step requires `rust`
+
+```bash
```
Comment (duanyytop) [inline README.md]: Adding `git clone git@github.com:nervosnetwork/molecule.git` will be more better

File: src/cli.ts [hunk 2]
```diff
@@ -3,35 +3,38 @@
...
-    }
+const printNormalizedSchema = (filePath: string) => {
+  const normalizedSchema = Schema.fromFile(filePath).getNormalizedSchema()
+  console.info(JSON.stringify(normalizedSchema, null, 2))
+  process.exit(0)
+}
 
-    if (!fs.existsSync(schemaOrPath)) {
+const outputGenerator = () => {
+  const schameOrPath = arg0
```
Comment (duanyytop) [inline src/cli.ts]: `const schemaOrPath = arg0`

Comment (duanyytop) [inline src/cli.ts]:
```suggestion
  const schemaOrPath = arg0
```

File: src/cli.ts [hunk 3]
```diff
@@ -3,35 +3,38 @@
...
+  const outputFile = arg1
+  try {
+    if (!fs.existsSync(schameOrPath)) {
       try {
-        JSON.parse(schemaOrPath)
+        JSON.parse(schameOrPath)
```
Comment (duanyytop) [inline src/cli.ts]:
```suggestion
        JSON.parse(schemaOrPath)
```

File: src/cli.ts [hunk 4]
```diff
@@ -3,35 +3,38 @@
...
       } catch {
         throw new Error(`Expect schema to be a JSON`)
       }
     }
-    const normalizeScript = fs.existsSync(schemaOrPath)
-      ? `const normalizedSchema = Schema.fromFile('${schemaOrPath}').getNormalizedSchema()`
-      : `const normalizedSchema = new Schema(${schemaOrPath}).getNormalizedSchema()`
+    const normalizeScript = fs.existsSync(schameOrPath)
```
Comment (duanyytop) [inline src/cli.ts]:
```suggestion
    const normalizeScript = fs.existsSync(schemaOrPath)
```

File: src/cli.ts [hunk 5]
```diff
@@ -3,35 +3,38 @@
...
+      ? `const normalizedSchema = Schema.fromFile('${schameOrPath}').getNormalizedSchema()`
```
Comment (duanyytop) [inline src/cli.ts]:
```suggestion
      ? `const normalizedSchema = Schema.fromFile('${schemaOrPath}').getNormalizedSchema()`
```

File: src/cli.ts [hunk 6]
```diff
@@ -3,35 +3,38 @@
...
+      : `const normalizedSchema = new Schema(${schameOrPath}).getNormalizedSchema()`
```
Comment (duanyytop) [inline src/cli.ts]:
```suggestion
      : `const normalizedSchema = new Schema(${schemaOrPath}).getNormalizedSchema()`
```

File: README.md (hunk 1 above)
Comment (Keith-CY) [reply README.md]: Why is `git clone` better than `cargo install`?

And this is consisitent with https://github.com/driftluo/moleculec-go/

File: README.md [hunk 7]
```diff
@@ -1,6 +1,161 @@
...
+```js
+const normalizedSchema = {
+  name: 'Bytes',
+  type: 'fixvec',
+  item: {
+    type: 'byte',
+  },
+}
+const data = ['0x01', '0x02']
+const molecule = new MoleculeJavaScript.Molecule(normalizedSchema)
```
Comment (duanyytop) [inline README.md]: Where `MoleculeJavaScript ` is from ?

File: README.md (hunk 1 above)
Comment (duanyytop) [reply README.md]: No. I mean add a shell comman not repalce the `cargo` command

File: README.md (hunk 7 above)
Comment (Keith-CY) [reply README.md]: > Where `MoleculeJavaScript ` is from ?

It's exposed by `webpack` bundled file, namely the `lib/index.js`

Comment (duanyytop) [reply README.md]: I suggest we should add description about this usage.

File: README.md (hunk 1 above)
Comment (Keith-CY) [reply README.md]: You mean there should be two ways to enable the `moleculec`?

Comment (duanyytop) [reply README.md]:
```
$ git clone git@github.com:nervosnetwork/molecule.git
$ cargo install moleculec
$ moleculec --language - --format json --schema-file ./types.mol > ./types.json
```

Comment (Keith-CY) [reply README.md]: > ```
> $ git clone git@github.com:nervosnetwork/molecule.git
> $ cargo install moleculec
> $ moleculec --language - --format json --schema-file ./types.mol > ./types.json
> ```

What does `git clone` do here?

Comment (Keith-CY) [reply README.md]: I've got your point. `moleculec` could be installed outside the origin repo.

Comment (duanyytop) [reply README.md]: Where `moleculec ` is from ?

Comment (Keith-CY) [reply README.md]: > Where `moleculec ` is from ?

From `cargo`, it's the `npm` for `rust`

Comment (duanyytop) [reply README.md]: OK.  I see. I just think `moleculec ` is from github repo in the past.

File: README.md (hunk 7 above)
Comment (Keith-CY) [reply README.md]: > I suggest we should add description about this usage.

The info of umd js package has been updated in https://github.com/Keith-CY/molecule-javascript/pull/65/commits/5027fbbe64c72f664427a88aeb4ddf87c2117d68

Comment (duanyytop) [reply README.md]: I mean we should add the desription which includes how to generate this package and how to import into Browser. For me, I didn't know what the `MoleculeJavaScript` is .

I think [It's exposed by webpack bundled file, namely the lib/index.js] is OK.

Comment (Keith-CY) [reply README.md]: > I mean we should add the desription which includes how to generate this package and how to import into Browser. For me, I didn't know what the `MoleculeJavaScript` is .
> 
> I think "It's exposed by webpack bundled file, namely the lib/index.js" is OK.

For users, they should not care about the implementation detail. So we should not mention `webpack` in the docs. If we indicate the `webpack` here, we have to update it once we transfer to `parcel` or `rollup`.

Comment (duanyytop) [reply README.md]: I mean users do not care about the implementation detail, and I just suggest they should know what `MoleculeJavaScript ` is. 

If you think `The package is exposed as globalThis.MoleculeJavaScript` is OK, I think it is fine.

Comment (Keith-CY) [reply README.md]: > I mean users do not care about the implementation detail, and I just suggest they should know what `MoleculeJavaScript ` is.
> 
> If you think `The package is exposed as globalThis.MoleculeJavaScript` is OK, I think it is fine.

Yep, I think `globalThis.MoleculeJavaScript` is enough, it's the standard statement of the universal module(for both browsers and node.js) in javascript.

**Pull request merged**